File: app/src/main/java/io/neoterm/setup/connection/AssetsFileConnection.java
package io.neoterm.setup.connection;

import java.io.IOException;
import java.io.InputStream;

import io.neoterm.App;
import io.neoterm.setup.SetupHelper;
import io.neoterm.utils.AssetsUtils;

/**
 * @author kiva
 */

public class AssetsFileConnection extends OfflineConnection {
    @Override
    protected InputStream openInputStream() throws IOException {
        String arch = SetupHelper.INSTANCE.determineArchName();
        String fileName = "offline_setup/" + arch + ".zip";
        return AssetsUtils.INSTANCE.openAssetsFile(App.Companion.get(), fileName);
    }
}

File: app/src/main/java/io/neoterm/setup/connections/AssetsFileConnection.kt
package io.neoterm.setup.connections

import io.neoterm.App
import io.neoterm.setup.SetupHelper
import io.neoterm.utils.AssetsUtils
import java.io.IOException
import java.io.InputStream

/**
 * @author kiva
 */

class AssetsFileConnection : OfflineConnection() {
    @Throws(IOException::class)
    override fun openInputStream(): InputStream {
        val arch = SetupHelper.determineArchName()
        val fileName = "offline_setup/$arch.zip"
        return AssetsUtils.openAssetsFile(App.get(), fileName)
    }
}
